### PR TITLE
Configuration

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/COM-507

Changes:
- Added Config and ConfigDefinition.
- Implemented Retry and errors handling.
- Implemented API Request factory from Config

### CODE REVIEW

File: src/ConfigDefinition.php [hunk 1]
```diff
@@ -6,22 +6,138 @@
...
         /** @noinspection NullPointerExceptionInspection */
         $parametersNode
+            ->isRequired()
             ->children()
-                ->scalarNode('foo')
-                    ->defaultValue('baz')
-                ->end()
-            ->end()
-        ;
+                ->scalarNode('name')->isRequired()->cannotBeEmpty()->end()
```
Comment (michaljurecko) [inline src/ConfigDefinition.php:71]: Tento PR obsahuje: ConfigDefinition

File: src/Config.php [hunk 2]
```diff
@@ -10,9 +10,54 @@
 
 class Config extends BaseConfig
 {
+    public function getMaxTries(): int
```
Comment (michaljurecko) [inline src/Config.php:13]: Tento PR obsahuje: Config

File: src/Api/RequestFactory.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    public function create(): Request
+    {
+        $method = 'POST';
+        $uri = 'query?api-version=' . self::API_VERSION;
+        $headers = ['Content-Type' => 'application/json'];
+        return new Request($method, $uri, $headers, JsonHelper::encode($this->getBody()));
+    }
+
+    private function getBody(): array
```
Comment (michaljurecko) [inline src/Api/RequestFactory.php:31]: Tento PR obsahuje: generovanie API requestu z konfiguracie.

File: src/Extractor.php [hunk 4]
```diff
@@ -4,18 +4,28 @@
...
         $this->logger = $logger;
-        $this->client = $client;
+        $this->api = $api;
+        $this->requestFactory = $requestFactory;
+    }
+
+    public function extract(): void
+    {
+        $response = $this->api->send($this->requestFactory->create());
+        var_export($response->getBody()->getContents());
```
Comment (michaljurecko) [inline src/Extractor.php:29]: Tento PR obsahuje: Ziskanie API response, ktora sa zatial nespracovava ale vypise na vystup, vid `simple` test.

File: src/Api/Api.php [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+use GuzzleHttp\Exception\RequestException;
+use Retry\BackOff\ExponentialBackOffPolicy;
+use Retry\Policy\SimpleRetryPolicy;
+use Retry\RetryProxy;
+use Keboola\AzureCostExtractor\Exception\ExportRequestRetryException;
+use Keboola\AzureCostExtractor\Exception\ExportRequestException;
+use Keboola\Component\JsonHelper;
+use Keboola\Component\UserException;
+
+class Api
```
Comment (michaljurecko) [inline src/Api/Api.php:22]: Tato trieda riesi retry a spracovanie chyb pri API calls.

**Review by ondrajodas — APPROVED** [review]
ok

**Pull request merged**